tests/integ/modin/frame/test_equals.py

```diff
@@ -0,0 +1,45 @@
+#
```

Comment:
Can you add a test for comparing an int8 column with int16? Also, what happens if you compare an integer NUMBER column with a VARIANT column that happens to have integers?

Reply:
Added tests for both. 

> Also, what happens if you compare an integer NUMBER column with a VARIANT column that happens to have integers?

It will be False both in native pandas and snowpark pandas.